api/v1beta1/ibmpowervsmachine_types.go

```diff
@@ -62,6 +62,7 @@ type IBMPowerVSMachineSpec struct {
 
 	// Processors is Number of processors allocated.
 	// +optional
+	// +kubebuilder:validation:Pattern=^[0-9]\.[0-9][0-9]$
```

Comment:
I don't think its covering all the cases, ptal - https://regex101.com/r/qYe78r/1